src/migrator.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import Command from 'cli-engine-command'
+import Plugins from './plugins'
+import path from 'path'
+import fs from 'fs-extra'
+
+export default class extends Command {
+  plugins = new Plugins(this)
+
+  async run () {
+    let file = path.join(this.config.dirs.data, '..', 'heroku', 'plugins', 'plugins.json')
+    let pljson
+    try {
+      pljson = fs.readJSONSync(file)
+    } catch (err) {
+      this.debug(err.message)
+      return
+    }
+    for (let p of pljson) {
+      try {
+        // this.log('migrating V5 plugins')
+        if (p.tag === 'symlink') {
+          // this.log('install via symlink: ', p.name)
+          let pluginPath = fs.realpathSync(path.join(this.config.dirs.data, '..', 'heroku', 'plugins', 'node_modules', p.name))
+          await this.plugins.addLinkedPlugin(pluginPath)
+        } else {
+          // this.log('install via npm: ', p.name)
+          await this.plugins.install(p.name)
+        }
+      } catch (err) {
+        this.debug(err.message)
+      }
+    }
+    try {
+      // this.log('cleaning up..')
+      fs.removeSync(path.join(this.config.dirs.data, '..', 'heroku'))
```

Comment:
@RasPhilCo could you scope this tighter to just be the `plugins` directory?  At first I wanted to just do it for safety reasons, but now I think this would cause real issues because you would blow away the shim that gets people from v5 to v6 currently.

Also, could you only do this if all plugins have been successfully migrated?  I know that it is unlikely to work the second time, but it would help us in debugging issues.  When someone complains about missing commands we would be able to tell what they had, and have them run this explicitly with debugging to see what failed and why.